allennlp/training/regularizers/regularizer_applicator.py

```diff
@@ -0,0 +1,82 @@
+from typing import Dict
+import re
+import torch
+
+from allennlp.experiments.registry import Registry
+from allennlp.common.params import Params
+from allennlp.training.regularizer import Regularizer
+
+
+class RegularizerApplicator:
+    """
+    Applies regularizers to the parameters of a Module based on regex matches.
+    """
+    def __init__(self, all_regularizers: Dict[str, Regularizer]) -> None:
+        """
+        Parameters
+        ----------
+        all_regularizers : Dict[str, Callable[[torch.Tensor], None]], optional (default = None)
+            A dictionary mapping parameter regexes to regularizers to be applied to parameters
+            matching the regex.
+        """
+        self._regularizers = all_regularizers
+
+    def __call__(self, module: torch.nn.Module) -> torch.Tensor:
+        """
+        Parameters
+        ----------
+        module : torch.nn.Module, required
+            The module to regularize.
+        """
+        accumulator = 0.0
+        for parameter_regex, regularizer in self._regularizers.items():
+            for name, parameter in module.named_parameters():
+                if self.parameter_name_matches_regex(parameter_regex, name):
```

Comment:
What's the motivation for not in-lining this, with `if re.search(parameter_regex, name):`?